server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -309,10 +320,7 @@ void processPendingEvents() throws Exception {
   }
 
   private void cleanupFlushEventData(ShuffleDataFlushEvent event) {
-    ShuffleBuffer shuffleBuffer = event.getShuffleBuffer();
-    if (shuffleBuffer != null) {
-      shuffleBuffer.clearInFlushBuffer(event.getEventId());
-    }
+    event.doCleanup();
     if (shuffleServer != null) {
```

Comment:
Could we put the logic into `doCleanup`?

Reply:
Yes. But the `shuffleBufferManager` is not a member variable for `ShuffleBuffer`

Reply:
> Yes. But the `shuffleBufferManager` is not a member variable for `ShuffleBuffer`

We can add this logic as a cleanup callback. It's ok although ShuffleBufferManager is not a member variable for ShuffleBuffer.

Reply:
Do you mean that we need to create a callback of `ShuffleBufferManager#releaseMemory` in `shuffleBuffer`  for the usage of event's `cleanupCallback`

Reply:
Sorry. Could you help check this question again?

Reply:
> Do you mean that we need to create a callback of `ShuffleBufferManager#releaseMemory` in `shuffleBuffer` for the usage of event's `cleanupCallback`

No.  I mean we could add a cleanupCallback for ShuffleFlushDataEvent in
https://github.com/apache/incubator-uniffle/blob/9a552b42f3b5073600b995d0da38204c2fe65e58/server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java#L211

Reply:
like 
```
  ShuffleDataFlushEvent event =
        buffer.toFlushEvent(
            appId,
            shuffleId,
            startPartition,
            endPartition,
            () -> bufferPool.containsKey(appId),
            shuffleFlushManager.getDataDistributionType(appId));
event.addCleanupCallback(
 xxxxreleaseMemoryxxxx
)
```

Reply:
Got your thought. Thanks for your advice.